lib/ansible/galaxy/token.py

```diff
@@ -54,18 +58,26 @@ def __init__(self, access_token=None, auth_url=None, validate_certs=True, client
         self.access_token = access_token
         self.auth_url = auth_url
         self._token = None
+        self._expires_at = None
         self.validate_certs = validate_certs
         self.client_id = client_id
         if self.client_id is None:
             self.client_id = 'cloud-services'
 
+    @property
+    def expired(self):
+        # expire early to handle refreshing proactively
+        return datetime.now(tz=timezone.utc) > self._expires_at - timedelta(minutes=AH_TOKEN_EXPIRES_N_MIN_EARLY)
+
     def _form_payload(self):
         return 'grant_type=refresh_token&client_id=%s&refresh_token=%s' % (self.client_id,
                                                                            self.access_token)
 
     def get(self):
-        if self._token:
+        if self._token and not self.expired:
             return self._token
+        if self._token and self.expired:
```

Comment:
The second clause is redundant, plus the log message gives it away. This is probably unnecessary.
```suggestion
        if self._token:
```